### PR TITLE
Added txtpen support and docs

### PR DESCRIPTION
**Pull request by rickyhan**

### CODE REVIEW

**Comment by moorereason** [comment]
It looks like you wrote txtpen and released it less than 24 hours ago, correct?  Why would we bake in support for a product that's not even a day old?

Everything in this PR can be done with regular site params and partials.  I bet we'd accept a guide in the docs, but I'm :-1: on adding built-in support for txtpen.  It would need to get enormously popular for that.

**Comment by rickyhan** [comment]
I understand your concerns and will remove the internal templates.

txtpen is currently the only platform that allows highlights and inline commenting. It would be great if you could put it in the docs and let new hugo users know about this functionality.

File: docs/content/overview/quickstart.md [hunk 1]
```diff
@@ -471,6 +471,8 @@ $ hugo server --theme=hugo_theme_robust
 
 ## Step 10. Integrate Disqus
 
+txtpen enables text highlights and in-text comments similar to Medium. To enable txtpen, set `txtpenShortname` in the `config.toml`.
```
Comment (moorereason) [inline docs/content/overview/quickstart.md]: Don't add txtpen to the quick start guide.

File: docs/content/overview/introduction.md [hunk 2]
```diff
@@ -133,6 +133,7 @@ Hugo boasts the following features:
 ### Additional Features
 
   * Integrated [Disqus](https://disqus.com/) comment support
+  * [txtpen](https://txtpen.com/) highlights and in-text commenting support
```
Comment (rdwatters) [inline docs/content/overview/introduction.md]: Not a knock on Txtpen by any means (very cool project, btw), but I don't think it's going to have the same draw as our internal templates for the most common use cases (i.e., GA and Disqus).

Comment (rickyhan) [inline docs/content/overview/introduction.md]: I appreciate it. Thanks for your kind words. Maybe it'll become common in a year or so to warrant an internal template :smile:

File: docs/content/overview/quickstart.md (hunk 1 above)
Comment (rickyhan) [inline docs/content/overview/quickstart.md]: Removed

**Comment by rickyhan** [comment]
Thank you. Really appreciate it :smile:

**Comment by rickyhan** [comment]
Following up on this. What else needs to be done :question: 

Jekyll has added txtpen to its [docs](https://jekyllrb.com/docs/extras/#web-highlights-and-commenting).

**Comment by digitalcraftsman** [comment]
I'm fine with adding it as an alternative to Disqus. But I've to agree with @moorereason that the service is quite new and doesn't justify the creation of an internal template until it hits the critical mass.

tl;dr: this PR can be merged.

**Comment by rickyhan** [comment]
@digitalcraftsman I have rebased HEAD~6 and squashed the commits

**Comment by digitalcraftsman** [comment]
I actually meant that only one commits remains after the rebase with all your additions. Nonetheless, I merged your pull request in 7cdc244a.
